src/pages/workspaces/workspace/Analyses.js

```diff
@@ -401,7 +405,8 @@ const Analyses = _.flow(
         div({ style: { color: colors.dark(), fontSize: 24, fontWeight: 600 } }, ['Your Analyses']),
         h(ButtonOutline, {
           style: { marginLeft: '1.5rem' },
-          onClick: () => setCreating(true),
+          //TODO: azure should eventually leverage create modal
+          onClick: () => !!googleProject ? setCreating(true) : setAzureCreating(true),
```

Comment:
So right now the issue isn't that there is unshared code being duplicated. Right now the flow for now azure environments is

`setCreating` opens the modal to create an analysis file, and after that, it opens the environment create modal.

For azure, we do not support analysis files, so we skip the intermediate step entirely. It would be possible to take the same approach cromwell does and have a banner in the `setCreating` modal, which opens the environment create modal directly. However, we plan to support analysis file creation for azure eventually, so we'd rather bunch up that work and asset creation.